department-of-veterans-affairs/va.gov-cms
# VACMS-2743: Fix image cropping issues.

## Description

See #2743. 

The only way I can think of to effectively determine whether the platform is generating cropped images correctly is with visual regression testing.  So after some research and playing briefly with a couple of different options, and thinking about the options available in deployment (for instance, `imagick` doesn't seem to be installed on Tugboat, and it might not be present on BRD), I figured [cypress-visual-regression](https://www.npmjs.com/package/cypress-visual-regression) was likely to be the best path forward.

I'm going to write the tests first and then try to find the simplest change possible that will get the tests to pass.

There are several cases I'm concerned about:
1. when a new node is saved and the image is freshly uploaded and cropped
2. when the derivative is subsequently deleted
3. when a node is edited and the image is freshly uploaded and cropped
4. when a node is edited and the existing image is recropped

(I might think of more as I dig deeper)

Of these cases, I think the only one that does not currently work is # 1, but it's also the most important 😛  This will be confirmed or disproven with tests.

## Testing done
Forthcoming.

## Screenshots
Forthcoming.  Boy, are these forthcoming.

## QA steps
As a user with permission to create new image media:
1. Create a new image media (medium? 🤔 )
2. Crop the image in some subjectively satisfying way
3. Verify that:
    - [x] the image's `2_1_medium_thumbnail` derivative reflects the crop you've selected.

### Definition of Done

### Select Team for PR review

### Is this PR blocked by another PR?

### Does this PR need review from a Product Owner

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
